storage/spanstore/reader/reader.go

```diff
@@ -316,17 +320,19 @@ func (s *SpanReader) queryPartitionList(ctx context.Context) ([]schema.Partition
 		if err != nil {
 			return err
 		}
-		defer res.Close()
-
-		result = make([]schema.PartitionKey, 0, res.TotalRowCount())
-		for res.NextResultSet(ctx, "part_date", "part_num", "is_active") {
-			for res.NextRow() {
-				part := schema.PartitionKey{}
-				if err = res.ScanWithDefaults(&part.Date, &part.Num, &part.IsActive); err != nil {
-					return err
-				}
-				result = append(result, part)
+		defer func() {
+			_ = res.Close()
+		}()
+		if !res.NextResultSet(ctx, "part_date", "part_num", "is_active") {
+			return ErrNothingResultSets
+		}
+		result = make([]schema.PartitionKey, 0, res.ResultSetCount())
```

Comment:
this slice size should be number of rows, not number of result sets

Reply:
Ack